scenarios/kubernetes_e2e_test.py

```diff
@@ -157,86 +156,87 @@ def test_local(self):
 
     def test_check_leaks_unset(self):
         """Ensure --check-leaked-resources=true sends flag to kubetest."""
-        args = self.parser.parse_args(['--mode=local'])
+        args = kubernetes_e2e.parse_args(['--mode=local'])
         with Stub(kubernetes_e2e, 'check_env', self.fake_check_env):
             kubernetes_e2e.main(args)
-            self.assertIn('--check-leaked-resources=false', self.callstack[-1])
+            self.assertNotIn('--check-leaked-resources', self.callstack[-1])
             self.assertEquals('false', self.envs.get('FAIL_ON_GCP_RESOURCE_LEAK'))
 
-    def test_extract_multiple(self):
-        """Ensure multiple --extract flags are accepted."""
-        args = self.parser.parse_args(['--extract=a', '--extract=b'])
-        self.assertEquals(['a', 'b'], args.extract)
-
-    def test_upgrade_args(self):
-        """Ensure upgrade_args flags are passed to kubetest."""
-        args = self.parser.parse_args(['--upgrade_args=foo bar'])
-        with Stub(kubernetes_e2e, 'check_env', self.fake_check_env):
-            kubernetes_e2e.main(args)
-            self.assertIn('--upgrade_args=foo bar', self.callstack[-1])
-
-    def test_extract_args(self):
-        """Ensure extract flags are passed to kubetest."""
-        args = self.parser.parse_args(['--extract=foo', '--extract=bar'])
-        with Stub(kubernetes_e2e, 'check_env', self.fake_check_env):
-            kubernetes_e2e.main(args)
-            self.assertIn('--extract=foo', self.callstack[-1])
-            self.assertIn('--extract=bar', self.callstack[-1])
-
-    def test_updown(self):
-        """Make sure local mode is fine overall."""
-        args = self.parser.parse_args(['--mode=local', '--up=false'])
-        self.assertEqual(args.mode, 'local')
+    def test_migrated_kubetest_args(self):
+        migrated = [
+            '--stage=a-stage',
+            '--stage-suffix=panda',
+            '--random-flag', 'random-value',
+            '--multiple-federations',
+            'arg1', 'arg2',
+            '--federation',
+            '--kubemark',
+            '--extract=this',
+            '--extract=that',
+            '--perf-tests',
+            '--deployment=yay',
+            '--save=somewhere',
+            '--skew',
+            '--publish=location',
+            '--timeout=42m',
+            '--upgrade_args=ginkgo',
+            '--up=true',
+            '--down=false',
+            '--check-leaked-resources=true',
+            '--charts',
+        ]
+        args = kubernetes_e2e.parse_args(['--mode=docker'] + migrated + ['--test=false'])
+        self.assertEquals(migrated, args.kubetest_args)
         with Stub(kubernetes_e2e, 'check_env', self.fake_check_env):
             kubernetes_e2e.main(args)
-
         lastcall = self.callstack[-1]
-        self.assertNotIn('--up', lastcall)
-        self.assertIn('--down', lastcall)
+        for arg in migrated:
+            self.assertIn(arg, lastcall)
```

Comment:
Can we ensure the order is still the same in the end as well? It can be a simpler test, since extract order matters.